app/models/course.rb

```diff
@@ -149,6 +149,27 @@ def application_forms
       .distinct
   end
 
+  def self.find_courses_in_previous_cycle(provider)
+    previous_cycle
+        .joins(:course_options)
```

Comment:
Why do we join to `:course_options` here and in the other method?

Reply:
Ah I see, in that case we should perhaps indicate that in the method or scope name.
eg.

```ruby
def courses_with_options_in_cycle(provider: provider, cycle: cycle)
...
```

Reply:
Ah I see, the refactoring into the scope `:with_course_options_run_by_provider` really clarifies this now :+1: